File: java-to-kotlin-starter-guide/src/main/java/me/sungbin/lec04/Lec04Main.java
package me.sungbin.lec04;

public class Lec04Main {
    public static void main(String[] args) {
        JavaMoney javaMoney1 = new JavaMoney(2_000L);
        JavaMoney javaMoney2 = new JavaMoney(1_000L);

        if (javaMoney1.compareTo(javaMoney2) > 0) {
            System.out.println("Money1이 Money2보다 큽니다.");
        }

        JavaMoney money1 = new JavaMoney(1_000L);
        JavaMoney money2 = money1;
        JavaMoney money3 = new JavaMoney(1_000L);

        System.out.println(money1 == money2);
        System.out.println(money1 == money3);
        System.out.println(money1.equals(money3));

        System.out.println(javaMoney1.plus(javaMoney2));
    }
}

File: java-to-kotlin-starter-guide/src/main/kotlin/me/sungbin/lec04/Lec04Main.kt
package me.sungbin.lec04

fun main() {
    val javaMoney1 = JavaMoney(2_000)
    val javaMoney2 = JavaMoney(1_000)

    if (javaMoney1 > javaMoney2) {
        println("Money1이 Money2보다 큽니다.")
    }

    val money1 = JavaMoney(1_000)
    val money2 = money1
    val money3 = JavaMoney(1_000)

    println(money1 === money2)
    println(money1 === money3)
    println(money1 == money3)

    if (fun1() || fun2()) {
        println("본문")
    }

    if (fun2() && fun1()) {
        println("본문")
    }

    val kotlinMoney1 = Money(1_000)
    val kotlinMoney2 = Money(2_000)

    println(kotlinMoney1 + kotlinMoney2)
}

fun fun1(): Boolean {
    println("fun1")

    return true
}

fun fun2(): Boolean {
    println("fun2")

    return false
}